sale_commission/model/sale_order.py

```diff
@@ -0,0 +1,218 @@
+        """
+
+        commission = self.commission_id
+        if commission.id:
+            agent_commission = self.agent_id.commission
+            if self.agent_id and commission.sections:
+                if commission.id != agent_commission.id:
+                    return {
+                        "warning": {
+                            "title": _('Fee installments!'),
+                            "message": _(
+                                "Selected commission has been assigned "
+                                "by sections and it does not match "
+                                "the one defined to the selected agent."
+                                "These sections shall apply only on this bill."
+                            )
+                        }
+                    }
+
+
+class sale_order(models.Model):
+    """Include commission behavior in sale order model"""
+
+    _inherit = "sale.order"
+
+    sale_agent_ids = fields.One2many(
+        "sale.order.agent",
+        "sale_id",
+        string="Agents",
+        states={"draft": [("readonly", False)]}
+    )
+
+    # XXX: It doesn't work with Odoo v8.0 and I think it is not necessary
+    # def create(self, cr, uid, values, context=None):
+    #     """Add sale order reference on sale.order.agent
+    #     """
+    #     if context is None:
+    #         context = {}
+    #     agent_pool = self.pool.get('sale.order.agent')
+    #     res = super(sale_order, self).create(
+    #          cr, uid, values, context=context)
+    #     if 'sale_agent_ids' in values:
+    #         for sale_order_agent in values['sale_agent_ids']:
+    #             agent_pool.write(
+    #           cr, uid, sale_order_agent[1], {'sale_id': res})
+    #     return res
+
+    # XXX: It doesn't work with Odoo v8.0 and I think it is not necessary
+    # def write(self, cr, uid, ids, values, context=None):
+    #     """Rebuild sale.order reference on sale.order.agent
+    #     """
+    #     if context is None:
+    #         context = {}
+    #     agent_pool = self.pool.get('sale.order.agent')
+    #     if 'sale_agent_ids' in values:
+    #         for sale_order_agent in values['sale_agent_ids']:
+    #             for id in ids:
+    #                 if sale_order_agent[2]:
+    #                     sale_order_agent[2]['sale_id'] = id
+    #                 else:
+    #                     agent_pool.unlink(
+    #           cr, uid, sale_order_agent[1], context=context)
+    #     return super(sale_order, self).write(
+    #           cr, uid, ids, values, context=context)
+
+    def onchange_partner_id(self, cr, uid, ids, part, context=None):
+        """Agent id field will be changed according to new partner
+        """
+        if context is None:
+            context = {}
+        sale_agent_ids = []
+        res = super(sale_order, self).onchange_partner_id(
+            cr, uid, ids, part, context=context
+        )
+        if res.get('value', False) and part:
```

Comment:
if value is not in res, `res.get('value')` will return `None` so you can remove `, False`